catalyst/contrib/scheduler/onecycle.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import _LRScheduler
+
+from catalyst.dl.utils import get_optimizer_momentum, set_optimizer_momentum
+
+
+class OneCycleLR(_LRScheduler):
+    def __init__(
+            self,
+            optimizer: Optimizer,
+            num_epochs: int,
+            lr_range=(1.0, 0.005),
+            init_lr: float = None,
+            warmup_epochs: int = 0,
+            decay_epochs: int = 0,
```

Comment:
I think, it's better to use `warmup_factor` and `decay_factor` (not sure about the factor suffix, btw, we need something better) and produce `warmup_epochs  = num_epochs  * warmup_factor`, `decay_epochs  = num_epochs * decay_factor`

nevertheless, I understand, that we can keep both and use one, that is not None
and do not forget, that this scheduler should work not only in `epoch` mode, but in the `batch` one too (cause the main feature of 1Cycle is lr & momentum change during the epoch, after each batch)

PS. just for code beauty it is much better to group variables by their meaning, something like
```
optimizer: Optimizer,
num_epochs: int,  # epochs block
warmup_epochs: int = 0,
decay_epochs: int = 0,
lr_range=(1.0, 0.005),   # lr block
init_lr: float = None,
momentum_range=(0.8, 0.99, 0.999),   # momentum block
init_momentum: float = None,
```